internal/providers/google/model.go

```diff
@@ -42,5 +44,25 @@ func Migrate(db *gorm.DB, logger logrus.FieldLogger) error {
 		"table_names": strings.TrimSpace(tableNames),
 	}).Info("migrating provider tables")
 
-	return db.AutoMigrate(tables...).Error
+	err := db.AutoMigrate(tables...).Error
+	if err != nil {
+		return err
+	}
+
+	err = model.AddForeignKey(db, logger, &model.ClusterModel{}, &GKEClusterModel{}, "ClusterID")
```

Comment:
IIRC foreign keys can be defined on structs as tags. Although visually not really appealing, it might be better to have all table definitions in a single place (the model).

Reply:
GORM doesn't create FKs based on struct tags, it uses the "foreighkey" struct tag solely for tracking the dependencies between models.

Reply:
Sad :(